ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-master.yaml

```diff
@@ -129,84 +129,10 @@ tests:
-        operator-sdk run bundle --timeout=10m --security-context-config restricted -n $WMCO_NS "$PREVIOUS_BUNDLE" \
-        || oc get csv -n $WMCO_NS |awk {'print $1'} | tail -n1 | xargs oc patch csv -n $WMCO_NS --type='json' \
-        -p="[{\"op\": \"replace\", \"path\": \"/spec/install/spec/deployments/0/spec/template/spec/containers/0/image\", \"value\": \"$PREVIOUS_OPERATOR\"}]"
-        sleep 10
-        # Delete the deployment which will then be recreated by the subscription controller with the correct image
-        oc delete deployment -n $WMCO_NS windows-machine-config-operator
-        # oc wait will immediately fail if the deployment does not exist yet, first retry until the deployment is
-        # created by the subscription controller
-        retries=0
-        while ! oc get -n $WMCO_NS deployment windows-machine-config-operator; do
-          if [[ $retries -eq 10 ]]; then
-            echo max retries hit
-            exit 1
-          fi
-          sleep 1m
-          retries=$((retries+1))
-        done
-        oc wait --timeout=10m --for condition=Available -n $WMCO_NS deployment windows-machine-config-operator
-      dependencies:
-      - env: PREVIOUS_BUNDLE
-        name: previous-bundle
-      - env: PREVIOUS_OPERATOR
-        name: previous-wmco
-      from: operator-sdk
-      resources:
-        requests:
-          cpu: 1000m
-          memory: 400Mi
-    - as: test-setup
-      cli: latest
-      commands: |-
-        #!/bin/bash
-        set -euo pipefail
-        export KUBE_SSH_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
-        oc create secret generic cloud-private-key --from-file=private-key.pem="${KUBE_SSH_KEY_PATH}" -n openshift-windows-machine-config-operator
-        make upgrade-test-setup
-      from: windows-machine-config-operator-test
-      resources:
-        requests:
-          cpu: 1000m
-          memory: 400Mi
-    - as: upgrade
-      cli: latest
-      commands: |
-        export WMCO_NS=openshift-windows-machine-config-operator
-        operator-sdk run bundle-upgrade --timeout=10m --security-context-config restricted -n $WMCO_NS "$OO_BUNDLE"
-        sleep 5m
-        oc wait --timeout=5m --for condition=Available -n $WMCO_NS deployment windows-machine-config-operator
-      dependencies:
-      - env: OO_BUNDLE
-        name: wmco-bundle
-      from: operator-sdk
-      resources:
-        requests:
-          cpu: 1000m
-          memory: 400Mi
-    - as: test
-      cli: latest
-      commands: |-
-        #!/bin/bash
-        set -euo pipefail
-        export KUBE_SSH_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
-        make upgrade-test
-      from: windows-machine-config-operator-test
-      resources:
-        requests:
-          cpu: 1000m
-          memory: 400Mi
+    - ref: windows-e2e-operator-test-upgrade-install
+    - ref: windows-e2e-operator-test-upgrade-setup
+    - ref: windows-e2e-operator-test-upgrade-upgrade
+    - ref: windows-e2e-operator-test-upgrade-test
```

Comment:
consider making this a `chain` so we don't have to list out all 4 steps in each config file

Reply:
Yes, definitely doable. I leaned towards explicit and separate steps for more control and readability. 

Happy to make the adjustment. @openshift/openshift-team-windows-containers  Please chime in.

Reply:
In any case, the steps can always be referenced individually if needed. Will push an upgrade introducing the chain.  Thanks for calling this out  @saifshaikh48